File: Lab_6/Codes/RGB_TO_GRAYSCALE.m
% Read the RGB image
rgbImage = imread('flower.jpg');

% Method 1: Using the Formula
% Convert to grayscale using the weighted sum formula
grayscaleImage_formula = 0.2989 * rgbImage(:,:,1) + 0.5870 * rgbImage(:,:,2) + 0.1140 * rgbImage(:,:,3);

% Display the original RGB image and grayscale image
figure;
subplot(1, 2, 1);
imshow(rgbImage);
title('Original RGB Image');
axis off;

subplot(1, 2, 2);
imshow(grayscaleImage_formula, []);
title('Grayscale Image (Formula)');
axis off;

% Method 2: Using MATLAB's built-in function
% Convert to grayscale using rgb2gray
grayscaleImage_builtin = rgb2gray(rgbImage);

% Display the grayscale image using built-in function
figure;
subplot(1, 2, 1);
imshow(rgbImage);
title('Original RGB Image');
axis off;

subplot(1, 2, 2);
imshow(grayscaleImage_builtin);
title('Grayscale Image (Built-in)');
axis off;

File: Lab_6/Codes/RGB_TO_GRAYSCALE.py
import numpy as np
import matplotlib.pyplot as plt
from PIL import Image

# Method 1: Using the Formula

# Read the RGB image
rgb_image = Image.open('flower.jpg').convert('RGB')
rgb_array = np.array(rgb_image)

# Convert to grayscale using the weighted sum formula
grayscale_image = 0.2989 * rgb_array[:, :, 0] + 0.5870 * rgb_array[:, :, 1] + 0.1140 * rgb_array[:, :, 2]

# Display the grayscale image
plt.figure(figsize=(8, 4))

plt.subplot(1, 2, 1)
plt.imshow(rgb_image)
plt.title('Original RGB Image')
plt.axis('off')

plt.subplot(1, 2, 2)
plt.imshow(grayscale_image, cmap='gray')
plt.title('Grayscale Image (Formula)')
plt.axis('off')

plt.tight_layout()
plt.show()


# Method 2: Using PIL

# Convert to grayscale using PIL
grayscale_image_pil = rgb_image.convert('L')

# Display the grayscale image
plt.figure(figsize=(8, 4))

plt.subplot(1, 2, 1)
plt.imshow(rgb_image)
plt.title('Original RGB Image')
plt.axis('off')

plt.subplot(1, 2, 2)
plt.imshow(grayscale_image_pil, cmap='gray')
plt.title('Grayscale Image (PIL)')
plt.axis('off')

plt.tight_layout()
plt.show()
